### PR TITLE
add support for hd archiving

### PR DESCRIPTION
**Pull request by msach22**

### CODE REVIEW

File: OpenTok/Archive.cs [hunk 1]
```diff
@@ -105,6 +105,7 @@ internal void CopyArchive(Archive archive)
             this.HasVideo = archive.HasVideo;
             this.HasAudio = archive.HasAudio;
             this.OutputMode = archive.OutputMode;
+			this.Resolution = archive.Resolution;
```
Comment (Nitrillo) [inline OpenTok/Archive.cs]: the indentation seems to be incorrect. could you use the same indentation? thanks!

File: OpenTok/OpenTok.cs [hunk 2]
```diff
@@ -259,15 +259,28 @@ public string GenerateToken(string sessionId, Role role = Role.PUBLISHER, double
...
                 throw new OpenTokArgumentException("Session not valid");
             }
             string url = string.Format("v2/project/{0}/archive", this.ApiKey);
             var headers = new Dictionary<string, string> { { "Content-type", "application/json" } };
-            var data = new Dictionary<string, object>() { { "sessionId", sessionId }, { "name", name }, { "hasVideo", hasVideo }, { "hasAudio", hasAudio }, { "outputMode", outputMode.ToString().ToLowerInvariant() } };
+            var data = new Dictionary<string, object>() { { "sessionId", sessionId }, { "name", name }, { "hasVideo", hasVideo }, { "hasAudio", hasAudio }, { "outputMode", outputMode.ToString().ToLowerInvariant() }, { "resolution", resolution } };
+
+			if (String.IsNullOrEmpty(resolution) && outputMode.Equals(OutputMode.COMPOSED))
+			{
+				resolution = "640x480";
```
Comment (Nitrillo) [inline OpenTok/OpenTok.cs]: shouldn't be this changing the value inside data?

Comment (Nitrillo) [reply OpenTok/OpenTok.cs]: could you add a test for this specific case, where we set the default resolution, checking that the resolution is set to the right value if is not added as an argument?

File: OpenTok/Archive.cs [hunk 3]
```diff
@@ -162,11 +163,15 @@ internal void CopyArchive(Archive archive)
          */
         public OutputMode OutputMode { get; set; }
 
-        /** 
+		/**
+		* The resolution of the Archive.
+		*/
+		public string Resolution { get; set; }
```
Comment (Nitrillo) [inline OpenTok/Archive.cs]: I'm not familiar with this code but, where is this property set?

File: OpenTok/OpenTok.cs (hunk 2 above)
Comment (Nitrillo) [reply OpenTok/OpenTok.cs]: Does it make sense to have public constants for the valid resolution values?

File: OpenTok/OpenTok.cs [hunk 4]
```diff
@@ -259,15 +259,28 @@ public string GenerateToken(string sessionId, Role role = Role.PUBLISHER, double
...
+			}
+			else if (String.IsNullOrEmpty(resolution) && outputMode.Equals(OutputMode.INDIVIDUAL))
+			{
+				data.Remove("resolution");
```
Comment (Nitrillo) [inline OpenTok/OpenTok.cs]: I would prefer positive logic, like adding the value to the dictionary when is needed.

File: OpenTok/Archive.cs (hunk 1 above)
Comment (msach22) [reply OpenTok/Archive.cs]: This is odd, it's not showing up like that on my editor.

**Comment by msach22** [comment]
This was accidentally merged. The changes have been reverted.

**Pull request merged**